Source/WebKit/UIProcess/Network/NetworkProcessProxyCocoa.mm

```diff
@@ -123,4 +128,27 @@
 
 #endif
 
+#if ENABLE(APPLE_PAY_REMOTE_UI_USES_SCENE)
+void NetworkProcessProxy::getWindowSceneAndBundleIdentifierForPaymentPresentation(WebPageProxyIdentifier webPageProxyIdentifier, CompletionHandler<void(const String&, const String&)>&& completionHandler)
```

Comment:
"get" is usually reserved for functions with out args, which this is not

Reply:
But, this is preexisting, so maybe?

Reply:
It's a pretty small API surface, so I'll just change it to `windowSceneAndBundleIdentifierForPaymentPresentation`

Reply:
Well... actually, that makes this call read very awkwardly? I'll just keep it as-is for now.

```cpp
    m_client.windowSceneAndBundleIdentifierForPaymentPresentation(webPageProxyID, [weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler)](const String& sceneIdentifier, const String& bundleIdentifier) mutable {
        if (!weakThis) {
            completionHandler(false);
            return;
        }

        if (!weakThis->m_authorizationPresenter) {
            completionHandler(false);
            return;
        }

        weakThis->m_authorizationPresenter->presentInScene(sceneIdentifier, bundleIdentifier, WTFMove(completionHandler));
    });
```

Reply:
Please convert the `weakThis` into a RefPtr right away, before you use it:
```
RefPtr protectedThis = weakThis.get();
if (!protectedThis) {
    completionHandler(false);
    return;
}
...
```

Reply:
Thanks for catching that. I'll fix it before landing. I should've read through the code a bit more carefully 🙂